From 7edd55da016c307f10f87b8e31a9b926ae706a9f Mon Sep 17 00:00:00 2001
From: dissent1 <be.dissent+github@gmail.com>
Date: Tue, 20 Jun 2017 11:23:20 +0300
Subject: [PATCH] clk: qcom: Boost all rpm clocks to max rate

This hack just boosts all RPM clocks to max values for better performance.

Based on https://source.codeaurora.org/quic/la/kernel/msm/commit/?h=linaro/release/qcomlt-4.9-next&id=f87198c71c88d5ce7303467c3ffcd54786d98f55
---
 drivers/clk/qcom/clk-rpm.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/clk/qcom/clk-rpm.c b/drivers/clk/qcom/clk-rpm.c
index 1762236515..06c76d9248 100644
--- a/drivers/clk/qcom/clk-rpm.c
+++ b/drivers/clk/qcom/clk-rpm.c
@@ -12,6 +12,7 @@
  * GNU General Public License for more details.
  */
 
+#include <linux/clk.h>
 #include <linux/clk-provider.h>
 #include <linux/err.h>
 #include <linux/export.h>
@@ -448,6 +449,7 @@ static int rpm_clk_probe(struct platform_device *pdev)
 	struct qcom_rpm *rpm;
 	struct clk_rpm **rpm_clks;
 	const struct rpm_clk_desc *desc;
+	struct clk *clk;
 
 	rpm = dev_get_drvdata(pdev->dev.parent);
 	if (!rpm) {
@@ -484,9 +486,14 @@ static int rpm_clk_probe(struct platform_device *pdev)
 		if (!rpm_clks[i])
 			continue;
 
-		ret = devm_clk_hw_register(&pdev->dev, &rpm_clks[i]->hw);
-		if (ret)
+		clk = devm_clk_register(&pdev->dev, &rpm_clks[i]->hw);
+		if (IS_ERR(clk)) {
+			ret = PTR_ERR(clk);
			goto err;
+		}
+
+		clk_set_rate(clk, INT_MAX);
+		clk_prepare_enable(clk);
 	}
 
 	ret = of_clk_add_hw_provider(pdev->dev.of_node, qcom_rpm_clk_hw_get,
